### PR TITLE
Add/remove staging workflows managers

### PR DESCRIPTION
**Pull request by dmarcoux**
After discussions this Monday about whether staging managers should be a role or not, we concluded that it is not.

The button to the index page:
staging_managers_button

The index page:
staging_managers_index

Add staging manager modal:
add_staging_manager_modal

Remove staging manager modal:
remove_staging_manager_modal

### CODE REVIEW

File: src/api/app/views/webui2/webui/staging_workflows_managers/_add_staging_manager_dialog.html.haml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                  %i.fa.fa-search
+                  %i.fas.fa-spinner.fa-spin.d-none
+              = text_field_tag 'staging_manager', '', required: true, placeholder: 'Type to autocomplete...', class: 'form-control'
+          = hidden_field_tag 'project', project
+          = hidden_field_tag 'webui2', true
+        .modal-footer
+          = render partial: 'webui2/shared/dialog_action_buttons'
+
+:javascript
+  $('#staging_manager').autocomplete({
```
Comment (coolo) [inline src/api/app/views/webui2/webui/staging_workflows_managers/_add_staging_manager_dialog.html.haml]: please put larger javascript functions in the asset - and just append a function call to :ready_function

**Comment by dmarcoux** [comment]
I'm looking into the specs... they all passed locally...

**Comment by dmarcoux** [comment]
[Add/Remove Staging Managers](https://trello.com/c/vFsRPIY0/894-add-remove-staging-managers)

**Comment by coolo** [comment]
What are staging managers supposed to do then? This sounds like a new user role to me - none we currently use, so I wonder. And how do you envision the new staging workflow with this new user role?

File: src/api/app/controllers/webui/staging_workflows_managers_controller.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
+class Webui::StagingWorkflowsManagersController < Webui::WebuiController
```
Comment (ChrisBr) [inline src/api/app/controllers/webui/staging_workflows_managers_controller.rb]: What about putting this in a module / namespace like:

```
Webui::StagingWorkflow::ManagersController
```

**Comment by ChrisBr** [comment]
> After discussions this Monday about whether staging managers should be a role or not, we concluded that it is not.

@dmarcoux can you maybe link the result of your discussion from monday? I would also be curious about the workflow and responsibilities ...

**Comment by dmarcoux** [comment]
@ChrisBr: We didn't want to re-use the relationship model for the roles/groups as they are for projects/packages, not for staging workflows. Instead of extending the relationship model, we want to have something specific to staging workflows.

**Comment by coolo** [comment]
But what's the point of this role? As you seem to redefine the staging process as we use it, some more background on your vision should be written down *somewhere*.

**Comment by dmarcoux** [comment]
@coolo: Staging managers are the users who will be handling a staging workflow by creating/deleting staging projects and then adding/removing the requests in the staging projects. Since this role is only for staging workflows, we didn't want to introduce this role at the project level (so in the users tab when on the project overview/show page).

**Comment by coolo** [comment]
And who is project maintainer in said staging projects?

**Comment by dmarcoux** [comment]
Staging projects won't have a project maintainer in the context of the staging workflow. We don't have the concept of a project maintainer in the staging projects. A staging manager manages everything in the staging workflow, this includes staging projects.

Project maintainers stay the same outside of the staging workflow context. This doesn't change. So a project maintainer for a project which happens to be a staging project in a staging workflow can still manage the project as usual (change the title, add some packages, etc...)

Do you see a need to further split the roles for staging workflows? We think that staging managers with rights to manage everything related to a staging workflows is enough.

**Comment by coolo** [comment]
If a staging manager manages staging projects, he has to have project maintainer rights in them.

**Comment by coolo** [comment]
or she

**Comment by coolo** [comment]
As a lot of things are blocked by this - can we step 2 steps back and stop reinventing the wheel and make the staging managers a group that is reviewer in the target project and be done with it?

For the moment we don't even need to create or modify the group, but just require it to be pre-created.

**Pull request closed without merging**